<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Case Study: Glycoproteomics • glycoverse</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Case Study: Glycoproteomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">glycoverse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.2.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/case-study-1.html">Case Study: Glycoproteomics</a></li>
    <li><a class="dropdown-item" href="../articles/case-study-2.html">Case Study: Glycomics</a></li>
  </ul>
</li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">Changelog</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Case Study: Glycoproteomics</h1>
            
      

      <div class="d-none name"><code>case-study-1.Rmd</code></div>
    </div>

    
    
<p>This vignette walks you through a complete glycoproteomics analysis
using <code>glycoverse</code>. We’ll explore the full spectrum of
glycoproteomics data analysis, from data loading and preprocessing to
statistical analysis and visualization. We’ll also dive into advanced
glycan structure analysis, including motif quantification and derived
trait analysis. Ready to dive in? Let’s go!</p>
<p><strong>Heads up:</strong> <code>glycoverse</code> is built on
<code>tidy</code> principles throughout. If you’re new to
<code>tidyverse</code> data analysis, we highly recommend checking out
Hadley Wickham’s excellent <a href="https://r4ds.hadley.nz" class="external-link">R for Data
Science</a>. Trust us, it’s worth the investment!</p>
<p>Quick readiness check:</p>
<ul>
<li>What’s a <code>tibble</code>?</li>
<li>How do you filter rows in a <code>tibble</code>?</li>
<li>What’s the modern alternative to <code>for</code> loops?</li>
<li>What’s the <code>%&gt;%</code> operator? Do we still need it?</li>
<li>What makes data “tidy”?</li>
</ul>
<div class="section level2">
<h2 id="tldr">TL;DR<a class="anchor" aria-label="anchor" href="#tldr"></a>
</h2>
<p>In case you’re in a hurry…</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load the packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glycoverse.github.io/glycoverse/">glycoverse</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Preprocess the data</span></span>
<span><span class="va">clean_exp</span> <span class="op">&lt;-</span> <span class="fu">auto_clean</span><span class="op">(</span><span class="va">real_experiment</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform PCA</span></span>
<span><span class="va">pca_res</span> <span class="op">&lt;-</span> <span class="fu">gly_pca</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">pca_res</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform differential expression analysis</span></span>
<span><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="fu">gly_limma</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span></span>
<span><span class="fu">get_tidy_result</span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform motif analysis</span></span>
<span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  lewis_by <span class="op">=</span> <span class="st">"dHex(??-?)Hex(??-?)[dHex(??-?)]HexNAc(??-"</span>,</span>
<span>  lewis_ax <span class="op">=</span> <span class="st">"Hex(??-?)[dHex(??-?)]HexNAc(??-"</span>,</span>
<span>  sia_lewis_ax <span class="op">=</span> <span class="st">"NeuAc(??-?)Hex(??-?)[dHex(??-?)]HexNAc(??-"</span></span>
<span><span class="op">)</span></span>
<span><span class="va">motif_exp</span> <span class="op">&lt;-</span> <span class="fu">quantify_motifs</span><span class="op">(</span><span class="va">clean_exp</span>, <span class="va">motifs</span><span class="op">)</span></span>
<span><span class="va">motif_anova_res</span> <span class="op">&lt;-</span> <span class="fu">gly_anova</span><span class="op">(</span><span class="va">motif_exp</span><span class="op">)</span></span>
<span><span class="fu">get_tidy_result</span><span class="op">(</span><span class="va">motif_anova_res</span>, <span class="st">"main_test"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Perform derived trait analysis</span></span>
<span><span class="va">trait_exp</span> <span class="op">&lt;-</span> <span class="fu">derive_traits</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span></span>
<span><span class="va">trait_anova_res</span> <span class="op">&lt;-</span> <span class="fu">gly_anova</span><span class="op">(</span><span class="va">trait_exp</span><span class="op">)</span></span>
<span><span class="fu">get_tidy_result</span><span class="op">(</span><span class="va">trait_anova_res</span>, <span class="st">"main_test"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="loading-the-packages">Loading the Packages<a class="anchor" aria-label="anchor" href="#loading-the-packages"></a>
</h2>
<p>We first load the <code>tidyverse</code> package, as usual.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">dplyr    </span> 1.1.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">readr    </span> 2.1.6</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">forcats  </span> 1.0.1     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">stringr  </span> 1.6.0</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">ggplot2  </span> 4.0.1     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tibble   </span> 3.3.0</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">lubridate</span> 1.9.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tidyr    </span> 1.3.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">purrr    </span> 1.2.0     </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span> masks <span style="color: #0000BB;">stats</span>::filter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>    masks <span style="color: #0000BB;">stats</span>::lag()</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use the conflicted package (<span style="color: #0000BB; font-style: italic;">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors</span></span></code></pre></div>
<p>Just like <code>tidyverse</code>, <code>glycoverse</code> is a
meta-package that loads a collection of specialized packages all at
once.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glycoverse.github.io/glycoverse/">glycoverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Attaching core glycoverse packages</span> ───────────────── glycoverse 0.2.0.<span style="color: #BB0000;">9000</span> ──</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyclean</span> 0.9.1          <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyparse</span> 0.5.3     </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glydet  </span> 0.6.5          <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyread </span> 0.8.2     </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glydraw </span> 0.0.0.<span style="color: #BB0000;">9000</span>     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyrepr </span> 0.9.0     </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyexp  </span> 0.10.4         <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glystats</span> 0.5.5     </span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glymotif</span> 0.11.2         <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyvis  </span> 0.4.1     </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ───────────────────────────────────────── glycoverse_conflicts() ──</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">glyclean</span>::<span style="color: #00BB00;">aggregate()</span>  masks <span style="color: #0000BB;">stats</span>::aggregate()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span>        masks <span style="color: #0000BB;">stats</span>::filter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">intersect()</span> masks <span style="color: #0000BB;">dplyr</span>::intersect(), <span style="color: #0000BB;">base</span>::intersect()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>           masks <span style="color: #0000BB;">stats</span>::lag()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">glyexp</span>::<span style="color: #00BB00;">select_var()</span>   masks <span style="color: #0000BB;">dplyr</span>::select_var()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">setdiff()</span>   masks <span style="color: #0000BB;">dplyr</span>::setdiff(), <span style="color: #0000BB;">base</span>::setdiff()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">setequal()</span>      masks <span style="color: #0000BB;">base</span>::setequal()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">union()</span>     masks <span style="color: #0000BB;">dplyr</span>::union(), <span style="color: #0000BB;">base</span>::union()</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use the conflicted package (<span style="color: #0000BB; font-style: italic;">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reading-the-data">Reading the Data<a class="anchor" aria-label="anchor" href="#reading-the-data"></a>
</h2>
<p>Data import is typically your first step in any analysis. For this
tutorial, we’ll use the <code>real_experiment</code> dataset that comes
with <code>glyexp</code>. This is a real-world N-glycoproteomics dataset
from 12 patients across four liver conditions: healthy controls (H),
hepatitis (M), cirrhosis (Y), and hepatocellular carcinoma (C), with 3
samples per condition.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">real_experiment</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Glycoproteomics Experiment</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Expression matrix: <span style="color: #0000BB;">12</span> samples, <span style="color: #0000BB;">4262</span> variables</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample information fields: <span style="color: #00BB00;">group</span> <span style="color: #0000BB;">&lt;fct&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Variable information fields: <span style="color: #00BB00;">peptide</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">peptide_site</span> <span style="color: #0000BB;">&lt;int&gt;</span>, <span style="color: #00BB00;">protein</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">protein_site</span> <span style="color: #0000BB;">&lt;int&gt;</span>, <span style="color: #00BB00;">gene</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">glycan_composition</span> <span style="color: #0000BB;">&lt;comp&gt;</span>, <span style="color: #00BB00;">glycan_structure</span> <span style="color: #0000BB;">&lt;struct&gt;</span></span></span></code></pre></div>
<p>For your own projects, the <code>glyread</code> package can import
data from virtually any mainstream glycoproteomics software—pGlyco3,
MSFragger-Glyco, Byonic, you name it. Each software has its own
dedicated import function. For instance, data from pGlyco3 with
pGlycoQuant quantification can be loaded using
<code>read_pglyco3_pglycoquant()</code>. Check out <a href="https://glycoverse.github.io/glyread/articles/glyread.html" class="external-link">Get
Started with glyread</a> for the full rundown.</p>
<p>The <code>real_experiment</code> object (like all outputs from
<code>glyread</code> functions) is an <code>experiment()</code> object.
If you’ve worked with <code>SummarizedExperiment</code> from
Bioconductor, think of <code>experiment()</code> as its tidy cousin.
Essentially, it’s a smart data container that manages three key
components:</p>
<ul>
<li>
<strong>Expression matrix</strong>: quantitative data with samples
as columns and variables as rows</li>
<li>
<strong>Sample information</strong>: a tibble with sample metadata
(group, batch, demographics, etc.)</li>
<li>
<strong>Variable information</strong>: a tibble with feature
metadata (proteins, peptides, glycan compositions, etc.)</li>
</ul>
<p>You can get these data components by using
<code>get_expr_mat()</code>, <code>get_sample_info()</code>, and
<code>get_var_info()</code>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_expr_mat</span><span class="op">(</span><span class="va">real_experiment</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;             C1         C2           C3           H1         H2</span></span>
<span><span class="co">#&gt; GP1         NA         NA     10655.62 3.105412e+04         NA</span></span>
<span><span class="co">#&gt; GP2  414080036  609889761  78954431.49           NA   11724908</span></span>
<span><span class="co">#&gt; GP3  581723113  604842244 167889901.32 6.977076e+08  703566323</span></span>
<span><span class="co">#&gt; GP4 3299649335 2856490652 957651065.86 2.600523e+09 3229968280</span></span>
<span><span class="co">#&gt; GP5   30427048   34294394   6390129.81 5.159133e+07   37479075</span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_sample_info</span><span class="op">(</span><span class="va">real_experiment</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 12 × 2</span></span></span>
<span><span class="co">#&gt;    sample group</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> C1     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> C2     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> C3     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> H1     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> H2     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> H3     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> M1     M    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> M2     M    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> M3     M    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Y1     Y    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Y2     Y    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Y3     Y</span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_var_info</span><span class="op">(</span><span class="va">real_experiment</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,262 × 8</span></span></span>
<span><span class="co">#&gt;    variable peptide   peptide_site protein protein_site gene  glycan_composition</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;comp&gt;</span>            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> GP1      JKTQGK               1 P08185           176 SERP… <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">Ne</span>…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> GP2      HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">Ne</span>…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> GP3      HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> GP4      HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">Ne</span>…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> GP5      HJSTGCLR             2 P10909           291 CLU   <span style="color: #000000;">Hex</span>(6)<span style="color: #000000;">HexNAc</span>(5)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> GP6      HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">Ne</span>…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> GP7      HSHNJJSS…            6 P04196           345 HRG   <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GP8      HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">dH</span>…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GP9      HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(4)<span style="color: #000000;">HexNAc</span>(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> GP10     HSHNJJSS…            5 P04196           344 HRG   <span style="color: #000000;">Hex</span>(4)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">Ne</span>…</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4,252 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: glycan_structure &lt;struct&gt;</span></span></span></code></pre></div>
<p>For a deeper dive into <code>experiment()</code> objects, check out
<a href="https://glycoverse.github.io/glyexp/articles/glyexp.html" class="external-link">Get
Started with glyexp</a>.</p>
</div>
<div class="section level2">
<h2 id="data-preprocessing">Data Preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h2>
<p>Raw quantification data needs preprocessing before analysis—that’s
just a fact of life in omics. Typical steps include normalization,
missing value imputation, and batch effect correction. Rather than
making you implement these tedious steps manually, <code>glyclean</code>
provides a comprehensive preprocessing pipeline. Just call
<code>auto_clean()</code> on your <code>experiment()</code> object and
you’re good to go.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clean_exp</span> <span class="op">&lt;-</span> <span class="fu">auto_clean</span><span class="op">(</span><span class="va">real_experiment</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Normalizing data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> No QC samples found. Using default normalization method based on experiment type.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Experiment type is <span style="color: #0000BB;">"glycoproteomics"</span>. Using `normalize_median()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Normalization completed.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Removing variables with too many missing values</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> No QC samples found. Using all samples.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Applying preset <span style="color: #0000BB;">"discovery"</span>...</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Total removed: <span style="color: #0000BB;">24</span> (<span style="color: #0000BB;">0.56</span>%) variables.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Variable removal completed.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Imputing missing values</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> No QC samples found. Using default imputation method based on sample size.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample size &lt;= 30, using `impute_sample_min()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Imputation completed.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Aggregating data</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Aggregating to <span style="color: #0000BB;">"gfs"</span> level</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Aggregation completed.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Normalizing data again</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> No QC samples found. Using default normalization method based on experiment type.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Experiment type is <span style="color: #0000BB;">"glycoproteomics"</span>. Using `normalize_median()`.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Normalization completed.</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Correcting batch effects</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Batch column  not found in sample_info. Skipping batch correction.</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Batch correction completed.</span></span></code></pre></div>
<p>Your data is now analysis-ready!</p>
<p>Want to customize the preprocessing steps? See <a href="https://glycoverse.github.io/glyclean/articles/glyclean.html" class="external-link">Get
Started with glyclean</a> for the full toolkit.</p>
</div>
<div class="section level2">
<h2 id="statistical-analysis-and-visualization">Statistical Analysis and Visualization<a class="anchor" aria-label="anchor" href="#statistical-analysis-and-visualization"></a>
</h2>
<p>Time for the fun part—statistical analysis and visualization! We’ll
use <code>glystats</code> for the number crunching and
<code>glyvis</code> to make sense of the results visually.</p>
<p>Let’s kick off with PCA to get a bird’s-eye view of our data
structure.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_pca</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glyvis`</span></span></code></pre></div>
<p><img src="case-study-1_files/figure-html/unnamed-chunk-9-1.png" class="r-plt" width="384"></p>
<p><code>glyvis</code> isn’t designed for publication-ready figures, but
it’s perfect for quick exploratory visualization. Behind the scenes,
<code>plot_pca()</code> calls <code>gly_pca()</code> from
<code>glystats</code> and renders the results.</p>
<p>You can also break this down into separate steps:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca_res</span> <span class="op">&lt;-</span> <span class="fu">gly_pca</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glystats`</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">pca_res</span><span class="op">)</span>  <span class="co"># from `glyvis`</span></span></code></pre></div>
<p><img src="case-study-1_files/figure-html/unnamed-chunk-10-1.png" class="r-plt" width="384"></p>
<p>We actually recommend the two-step approach, since it gives you more
flexibility with the results. You can create custom <code>ggplot2</code>
visualizations for publications or extract the underlying data when
reviewers ask for it.</p>
<p><code>glystats</code> covers virtually all standard omics analyses.
All functions follow the same naming pattern:
<code>gly_xxx()</code>—think <code>gly_anova()</code>,
<code>gly_ttest()</code>, <code>gly_roc()</code>,
<code>gly_cox()</code>, <code>gly_wgcna()</code>, and so on. They all
take an <code>experiment()</code> object as their first argument.</p>
<p>The return format is consistent across all functions—a list with two
components:</p>
<ul>
<li>
<strong><code>tidy_result</code></strong>: cleaned-up tibbles in
tidy format. We’ve done the heavy lifting of organizing messy
statistical output for you.</li>
<li>
<strong><code>raw_result</code></strong>: the original statistical
objects. These are available when you need to dig deeper or perform
advanced analyses.</li>
</ul>
<p><code>glystats</code> provides two helper functions to get the tidy
result tibble and the raw result list from a glystats result object:
<code>get_tidy_result()</code> and <code>get_raw_result()</code>. Let’s
now see what the <code>samples</code> tibble looks like:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_tidy_result</span><span class="op">(</span><span class="va">pca_res</span>, <span class="st">"samples"</span><span class="op">)</span>  <span class="co"># many tibbles, so we specify one of them</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 144 × 4</span></span></span>
<span><span class="co">#&gt;    sample group    PC  value</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;fct&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> C1     C         1 -<span style="color: #BB0000;">21.7</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> C1     C         2  24.7 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> C1     C         3   1.89</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> C1     C         4   1.05</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> C1     C         5 -<span style="color: #BB0000;">11.5</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> C1     C         6  26.5 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> C1     C         7   5.81</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> C1     C         8   5.21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> C1     C         9 -<span style="color: #BB0000;">27.8</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> C1     C        10  -<span style="color: #BB0000;">7.01</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 134 more rows</span></span></span></code></pre></div>
<p>Notice the “group” column? That’s <code>glystats</code> being
helpful— it automatically pulls relevant metadata from your
<code>experiment()</code> object and includes it in the results wherever
it makes sense.</p>
<p>Back to that <code><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot()</a></code> magic we saw earlier. It
automatically recognizes different <code>glystats</code> result types
and plots accordingly— no manual specification needed. The plots won’t
win any beauty contests, but they’ll get your data insights across
fast.</p>
<p>The PCA clearly shows that our samples cluster nicely by
condition—always a good sign! Now let’s dive into differential
expression analysis using the tried-and-true <code>limma</code>
package.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="fu">gly_limma</span><span class="op">(</span><span class="va">clean_exp</span>, contrasts <span class="op">=</span> <span class="st">"H_vs_C"</span><span class="op">)</span>  <span class="co"># from `glystats`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of groups: <span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Groups: <span style="color: #0000BB;">"H"</span>, <span style="color: #0000BB;">"M"</span>, <span style="color: #0000BB;">"Y"</span>, and <span style="color: #0000BB;">"C"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Pairwise comparisons will be performed, with levels coming first as reference groups.</span></span>
<span><span class="fu">get_tidy_result</span><span class="op">(</span><span class="va">limma_res</span><span class="op">)</span>  <span class="co"># only one tibble here</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3,979 × 14</span></span></span>
<span><span class="co">#&gt;    variable protein glycan_composition      glycan_structure  protein_site gene </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;comp&gt;</span>                  <span style="color: #949494; font-style: italic;">&lt;struct&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V1       P08185  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">NeuAc</span>(2) <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(?…          176 SERP…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V2       P04196  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">NeuAc</span>(1) <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(?…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V3       P04196  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)         <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V4       P04196  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">NeuAc</span>(1) <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(?…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V5       P10909  <span style="color: #000000;">Hex</span>(6)<span style="color: #000000;">HexNAc</span>(5)         <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(…          291 CLU  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V6       P04196  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">NeuAc</span>(2) <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(?…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V7       P04196  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)         <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(…          345 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V8       P04196  <span style="color: #000000;">Hex</span>(5)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">dHex</span>(2)  <span style="color: #000000;">dHex</span>(??-?)<span style="color: #000000;">Hex</span>(??…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V9       P04196  <span style="color: #000000;">Hex</span>(4)<span style="color: #000000;">HexNAc</span>(3)         <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V10      P04196  <span style="color: #000000;">Hex</span>(4)<span style="color: #000000;">HexNAc</span>(4)<span style="color: #000000;">NeuAc</span>(1) <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(?…          344 HRG  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3,969 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 8 more variables: log2fc &lt;dbl&gt;, AveExpr &lt;dbl&gt;, t &lt;dbl&gt;, p_val &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   p_adj &lt;dbl&gt;, b &lt;dbl&gt;, ref_group &lt;chr&gt;, test_group &lt;chr&gt;</span></span></span></code></pre></div>
<p>Excellent! Now let’s identify significantly differentially expressed
glycoforms between HCC and healthy samples, then see what biological
pathways they’re involved in.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_sig_vars</span><span class="op">(</span><span class="va">limma_res</span>, p_adj_cutoff <span class="op">=</span> <span class="fl">0.05</span>, fc_cutoff <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">gly_enrich_go</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p><img src="case-study-1_files/figure-html/unnamed-chunk-13-1.png" class="r-plt" width="576"></p>
<p>And that’s it—pathway enrichment in just a few lines! Here we
filtered the experiment to keep only significant variables and then
performed pathway enrichment. As this operation is so common,
<code>glystats</code> provides a dedicated function for it:
<code>filter_sig_vars()</code>.</p>
<p>For the full statistical arsenal, check out <a href="https://glycoverse.github.io/glystats/articles/glystats.html" class="external-link">Get
Started with glystats</a> and <a href="https://glycoverse.github.io/glyvis/articles/glyvis.html" class="external-link">Get
Started with glyvis</a>.</p>
</div>
<div class="section level2">
<h2 id="advanced-motif-analysis">Advanced Motif Analysis<a class="anchor" aria-label="anchor" href="#advanced-motif-analysis"></a>
</h2>
<p>Up to now, we’ve covered standard glycoproteomics workflows. While
<code>glycoverse</code> certainly streamlines these analyses, it truly
shines when it comes to advanced glycan structure analysis.</p>
<p>Before diving into motifs, let’s get acquainted with
<code><a href="https://glycoverse.github.io/glyrepr/reference/glycan_structure.html" class="external-link">glyrepr::glycan_structure()</a></code> vectors.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">get_var_info</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">glycan_structure</span><span class="op">)</span></span>
<span><span class="co">#&gt; &lt;glycan_structure[3979]&gt;</span></span>
<span><span class="co">#&gt; [1] <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [2] <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)[<span style="color: #000000;">HexNAc</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [3] <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [4] <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [5] <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [6] <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [7] <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [8] <span style="color: #000000;">dHex</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">dHex</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [9] <span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; [10] <span style="color: #000000;">NeuAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)[<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">Hex</span>(??-?)]<span style="color: #000000;">Hex</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-?)<span style="color: #000000;">HexNAc</span>(??-</span></span>
<span><span class="co">#&gt; ... (3969 more not shown)</span></span>
<span><span class="co">#&gt; # Unique structures: 967</span></span></code></pre></div>
<p>Just like <code><a href="https://rdrr.io/r/base/integer.html" class="external-link">integer()</a></code> and <code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code>,
<code>glycan_structure()</code> is a specialized vector type. Some
software (like pGlyco3 and StrucGP) outputs structural information as
text strings. When you import this data using <code>glyread</code>, the
<code>glyparse</code> package automatically converts these strings into
proper <code>glycan_structure()</code> vectors and stores them in the
variable information tibble. Note that not all software provides
structural data—some only give compositions.</p>
<p>Fortunately, our example dataset includes structural information,
opening up a world of advanced analytical possibilities. Let’s explore
motif analysis.</p>
<p><strong>Quick note:</strong> The printed structures use
IUPAC-condensed notation, which we’ll also use for defining motifs
below. Don’t worry if it looks intimidating—we’ll include visual
diagrams to help. That said, if you’re planning to do serious structural
analysis, learning IUPAC-condensed notation is worth the investment.
Check out <a href="https://glycoverse.github.io/glyrepr/articles/iupac.html" class="external-link">this
guide</a> to get started—it’s easier than it looks!</p>
<p>Lewis antigen epitopes are common structural motifs found on
N-glycans. Ignoring linkage specificity, we can define three main Lewis
motif families:</p>
<p><img src="lewis_motifs.png"></p>
<p>Here’s how we express these motifs in IUPAC-condensed notation:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  lewis_by <span class="op">=</span> <span class="st">"dHex(??-?)Hex(??-?)[dHex(??-?)]HexNAc(??-"</span>,</span>
<span>  lewis_ax <span class="op">=</span> <span class="st">"Hex(??-?)[dHex(??-?)]HexNAc(??-"</span>,</span>
<span>  sia_lewis_ax <span class="op">=</span> <span class="st">"NeuAc(??-?)Hex(??-?)[dHex(??-?)]HexNAc(??-"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>A couple of important points:</p>
<ol style="list-style-type: decimal">
<li>We’re using generic monosaccharide names (“Hex” vs. “Glc”) to match
typical glycoproteomics data resolution</li>
<li>The “??-?” represents unknown linkages—a common limitation in mass
spectrometry data</li>
</ol>
<p>This level of structural ambiguity is typical in glycoproteomics. The
key is matching your motif definitions to your data’s resolution.</p>
<p>Here’s our research question: <strong>How many glycosites show
differential Lewis antigen expression across conditions?</strong>
Without <code>glycoverse</code>, this would be a nightmare to tackle
manually. Take a moment to imagine the pain of doing this by hand!</p>
<p>Now, the <code>glycoverse</code> solution:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motif_anova_res</span> <span class="op">&lt;-</span> <span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">quantify_motifs</span><span class="op">(</span><span class="va">motifs</span><span class="op">)</span> <span class="op">|&gt;</span>  <span class="co"># quantify these motifs</span></span>
<span>  <span class="fu">gly_anova</span><span class="op">(</span><span class="op">)</span>  <span class="co"># and perform ANOVA</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of groups: <span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Groups: <span style="color: #0000BB;">"H"</span>, <span style="color: #0000BB;">"M"</span>, <span style="color: #0000BB;">"Y"</span>, and <span style="color: #0000BB;">"C"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Pairwise comparisons will be performed, with levels coming first as reference groups.</span></span>
<span></span>
<span><span class="fu">get_tidy_result</span><span class="op">(</span><span class="va">motif_anova_res</span>, <span class="st">"main_test"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 828 × 13</span></span></span>
<span><span class="co">#&gt;    variable protein protein_site motif       gene  term     df    sumsq   meansq</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V1       A6NJW9            49 lewis_by    CD8B2 group     3 5.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span> 1.97<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V2       A6NJW9            49 lewis_ax    CD8B2 group     3 1.48<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span> 4.93<span style="color: #949494;">e</span><span style="color: #BB0000;">-32</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V3       A6NJW9            49 sia_lewis_… CD8B2 group     3 5.92<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span> 1.97<span style="color: #949494;">e</span><span style="color: #BB0000;">-31</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V4       O14786           150 lewis_by    NRP1  group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V5       O14786           150 lewis_ax    NRP1  group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V6       O14786           150 sia_lewis_… NRP1  group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V7       O43866           226 lewis_by    CD5L  group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V8       O43866           226 lewis_ax    CD5L  group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V9       O43866           226 sia_lewis_… CD5L  group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V10      O75437           244 lewis_by    ZNF2… group     3 0   <span style="color: #949494;"> </span>    0   <span style="color: #949494;"> </span>   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 818 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: statistic &lt;dbl&gt;, p_val &lt;dbl&gt;, p_adj &lt;dbl&gt;, post_hoc &lt;chr&gt;</span></span></span></code></pre></div>
<p><code>quantify_motifs()</code> transforms your data into a new
<code>experiment()</code> object. Instead of quantifying individual
glycans per glycosite, you now have motif abundances per glycosite
across samples. Since it’s still an <code>experiment()</code> object,
all <code>glystats</code> functions work seamlessly—including
<code>gly_anova()</code>.</p>
<p>Now we can answer our question using standard <code>tidyverse</code>
operations, since <code>motif_anova_res$tidy_result$main_test</code> is
just a regular tibble:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motif_anova_res</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">get_tidy_result</span><span class="op">(</span><span class="st">"main_test"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">motif</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># Groups:   motif [3]</span></span></span>
<span><span class="co">#&gt;   motif            n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> lewis_ax        27</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> lewis_by         9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> sia_lewis_ax    28</span></span></code></pre></div>
<p>Want the specific glycosites with significant Lewis a/x epitopes?
Easy:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motif_anova_res</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">get_tidy_result</span><span class="op">(</span><span class="st">"main_test"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">motif</span> <span class="op">==</span> <span class="st">"lewis_ax"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">protein</span>, <span class="va">protein_site</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 27 × 2</span></span></span>
<span><span class="co">#&gt;    protein protein_site</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> O75882           731</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P00734           143</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P00734           155</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P00738           241</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P01011           271</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P01042           294</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P01877           205</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P01877            92</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P02675           394</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P02679            78</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 17 more rows</span></span></span></code></pre></div>
<p>Here’s another common question: <strong>Which pathways are enriched
in proteins that carry Lewis a/x epitopes?</strong></p>
<p>For this analysis, we don’t need motif quantification—we just need to
know which proteins have these motifs.
<code><a href="https://glycoverse.github.io/glymotif/reference/add_motifs_int.html" class="external-link">glymotif::add_motifs_lgl()</a></code> is perfect for this.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">kegg_res</span> <span class="op">&lt;-</span> <span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">add_motifs_lgl</span><span class="op">(</span><span class="va">motifs</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_var</span><span class="op">(</span><span class="va">lewis_ax</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">gly_enrich_kegg</span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">kegg_res</span><span class="op">)</span></span></code></pre></div>
<p><img src="case-study-1_files/figure-html/unnamed-chunk-19-1.png" class="r-plt" width="576"></p>
<p><code>add_motifs_lgl()</code> adds three new TRUE/FALSE columns
(<code>lewis_by</code>, <code>lewis_ax</code>,
<code>sia_lewis_ax</code>) to the variable information.
<code>filter_var()</code> keeps only glycoforms with Lewis a/x epitopes.
Finally, <code>gly_enrich_kegg()</code> runs pathway enrichment on the
remaining proteins.</p>
<p><code>glymotif</code> has much more to offer beyond these examples.
Dive deeper with <a href="https://glycoverse.github.io/glymotif/articles/glymotif.html" class="external-link">Get
Started with glymotif</a>.</p>
</div>
<div class="section level2">
<h2 id="derived-trait-analysis">Derived Trait Analysis<a class="anchor" aria-label="anchor" href="#derived-trait-analysis"></a>
</h2>
<p>Let’s wrap up with derived traits—a clever analytical approach
developed by the N-glycomics community for glycome characterization.
Classic examples include:</p>
<ul>
<li>High-mannose glycan proportion</li>
<li>Core-fucosylation rate within complex glycans</li>
<li>Average sialylation per galactose residue</li>
</ul>
<p><code>glydet</code> adapts this concept for glycoproteomics by
treating each glycosite as its own mini-glycome. This enables
site-specific trait calculation and much richer biological insights.</p>
<p>Using <code>glydet</code> couldn’t be simpler:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trait_exp</span> <span class="op">&lt;-</span> <span class="fu">derive_traits</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glydet`</span></span>
<span><span class="va">trait_exp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Traitproteomics Experiment</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Expression matrix: <span style="color: #0000BB;">12</span> samples, <span style="color: #0000BB;">3864</span> variables</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample information fields: <span style="color: #00BB00;">group</span> <span style="color: #0000BB;">&lt;fct&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Variable information fields: <span style="color: #00BB00;">protein</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">protein_site</span> <span style="color: #0000BB;">&lt;int&gt;</span>, <span style="color: #00BB00;">trait</span> <span style="color: #0000BB;">&lt;chr&gt;</span>, <span style="color: #00BB00;">gene</span> <span style="color: #0000BB;">&lt;chr&gt;</span></span></span></code></pre></div>
<p>That’s it! Just like <code>quantify_motifs()</code>,
<code>derive_traits()</code> creates a new <code>experiment()</code>
object, but now with trait values per glycosite per sample.</p>
<p>The variable information shows what we’re working with:</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_var_info</span><span class="op">(</span><span class="va">trait_exp</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3,864 × 5</span></span></span>
<span><span class="co">#&gt;    variable protein protein_site trait gene </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V1       A6NJW9            49 TM    CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V2       A6NJW9            49 TH    CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V3       A6NJW9            49 TC    CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V4       A6NJW9            49 MM    CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V5       A6NJW9            49 CA2   CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V6       A6NJW9            49 CA3   CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V7       A6NJW9            49 CA4   CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V8       A6NJW9            49 TF    CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V9       A6NJW9            49 TFc   CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V10      A6NJW9            49 TFa   CD8B2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 3,854 more rows</span></span></span></code></pre></div>
<p>The “trait” column lists all the derived traits we can analyze.</p>
<p><code>glydet</code> comes with a comprehensive set of built-in
traits:</p>
<ul>
<li>
<strong><code>TM</code></strong>: Proportion of high-mannose
glycans</li>
<li>
<strong><code>TH</code></strong>: Proportion of hybrid glycans<br>
</li>
<li>
<strong><code>TC</code></strong>: Proportion of complex glycans</li>
<li>
<strong><code>MM</code></strong>: Average number of mannoses within
high-mannose glycans</li>
<li>
<strong><code>CA2</code></strong>: Proportion of bi-antennary
glycans within complex glycans</li>
<li>
<strong><code>CA3</code></strong>: Proportion of tri-antennary
glycans within complex glycans</li>
<li>
<strong><code>CA4</code></strong>: Proportion of tetra-antennary
glycans within complex glycans</li>
<li>
<strong><code>TF</code></strong>: Proportion of fucosylated
glycans</li>
<li>
<strong><code>TFc</code></strong>: Proportion of core-fucosylated
glycans</li>
<li>
<strong><code>TFa</code></strong>: Proportion of arm-fucosylated
glycans</li>
<li>
<strong><code>TB</code></strong>: Proportion of glycans with
bisecting GlcNAc</li>
<li>
<strong><code>SG</code></strong>: Average degree of sialylation per
galactose</li>
<li>
<strong><code>GA</code></strong>: Average degree of galactosylation
per antenna</li>
<li>
<strong><code>TS</code></strong>: Proportion of sialylated
glycans</li>
</ul>
<p>These represent the most widely used traits in glycomics
literature.</p>
<p>Let’s identify glycosites with significantly different
core-fucosylation levels (TFc) across conditions:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trait_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_var</span><span class="op">(</span><span class="va">trait</span> <span class="op">==</span> <span class="st">"TFc"</span><span class="op">)</span> <span class="op">|&gt;</span>  <span class="co"># from `glyexp`</span></span>
<span>  <span class="fu">gly_anova</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">get_tidy_result</span><span class="op">(</span><span class="st">"main_test"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of groups: <span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Groups: <span style="color: #0000BB;">"H"</span>, <span style="color: #0000BB;">"M"</span>, <span style="color: #0000BB;">"Y"</span>, and <span style="color: #0000BB;">"C"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Pairwise comparisons will be performed, with levels coming first as reference groups.</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 20 × 13</span></span></span>
<span><span class="co">#&gt;    variable protein protein_site trait gene   term     df     sumsq    meansq</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V247     P00450           397 TFc   CP     group     3 0.022<span style="text-decoration: underline;">1</span>    0.007<span style="text-decoration: underline;">35</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V401     P00738           211 TFc   HP     group     3 0.085<span style="text-decoration: underline;">3</span>    0.028<span style="text-decoration: underline;">4</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V415     P00738           241 TFc   HP     group     3 0.000<span style="text-decoration: underline;">784</span>  0.000<span style="text-decoration: underline;">261</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V457     P00748           249 TFc   F12    group     3 0.000<span style="text-decoration: underline;">548</span>  0.000<span style="text-decoration: underline;">183</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V709     P01591            71 TFc   JCHAIN group     3 0.077<span style="text-decoration: underline;">1</span>    0.025<span style="text-decoration: underline;">7</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V863     P01877            92 TFc   IGHA2  group     3 0.002<span style="text-decoration: underline;">02</span>   0.000<span style="text-decoration: underline;">673</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V919     P02679            78 TFc   FGG    group     3 0.003<span style="text-decoration: underline;">65</span>   0.001<span style="text-decoration: underline;">22</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V1115    P02765           176 TFc   AHSG   group     3 0.000<span style="text-decoration: underline;">094</span>1 0.000<span style="text-decoration: underline;">031</span>4</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V1227    P02790           240 TFc   HPX    group     3 0.062<span style="text-decoration: underline;">9</span>    0.021<span style="text-decoration: underline;">0</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V1353    P03952           494 TFc   KLKB1  group     3 0.002<span style="text-decoration: underline;">31</span>   0.000<span style="text-decoration: underline;">770</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> V1381    P04004            86 TFc   VTN    group     3 0.006<span style="text-decoration: underline;">50</span>   0.002<span style="text-decoration: underline;">17</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> V1661    P04278           396 TFc   SHBG   group     3 0.029<span style="text-decoration: underline;">9</span>    0.009<span style="text-decoration: underline;">98</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> V1675    P05090            98 TFc   APOD   group     3 0.017<span style="text-decoration: underline;">0</span>    0.005<span style="text-decoration: underline;">68</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> V1927    P06681           621 TFc   C2     group     3 0.054<span style="text-decoration: underline;">9</span>    0.018<span style="text-decoration: underline;">3</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> V2165    P0C0L4          <span style="text-decoration: underline;">1</span>328 TFc   C4A    group     3 0.017<span style="text-decoration: underline;">4</span>    0.005<span style="text-decoration: underline;">81</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> V2179    P0C0L4          <span style="text-decoration: underline;">1</span>391 TFc   C4A    group     3 0.000<span style="text-decoration: underline;">041</span>7 0.000<span style="text-decoration: underline;">013</span>9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> V2515    P19652           103 TFc   ORM2   group     3 0.064<span style="text-decoration: underline;">4</span>    0.021<span style="text-decoration: underline;">5</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> V2683    P25311           112 TFc   AZGP1  group     3 0.030<span style="text-decoration: underline;">5</span>    0.010<span style="text-decoration: underline;">2</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">19</span> V2697    P25311           128 TFc   AZGP1  group     3 0.000<span style="text-decoration: underline;">519</span>  0.000<span style="text-decoration: underline;">173</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">20</span> V2865    P43652            33 TFc   AFM    group     3 0.005<span style="text-decoration: underline;">47</span>   0.001<span style="text-decoration: underline;">82</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: statistic &lt;dbl&gt;, p_val &lt;dbl&gt;, p_adj &lt;dbl&gt;, post_hoc &lt;chr&gt;</span></span></span></code></pre></div>
<p>Once again, it’s just that straightforward.</p>
<p>This just scratches the surface of <code>glydet</code>’s
capabilities. The real power lies in defining custom traits tailored to
your research questions. Explore the possibilities in <a href="https://glycoverse.github.io/glydet/articles/glydet.html" class="external-link">Get
Started with glydet</a>.</p>
</div>
<div class="section level2">
<h2 id="whats-next">What’s Next?<a class="anchor" aria-label="anchor" href="#whats-next"></a>
</h2>
<p>This vignette has given you a taste of <code>glycoverse</code> in
action through a real-world glycoproteomics workflow. But we’ve barely
scratched the surface! Now that you’ve got the basics down, you’re ready
to unlock the full potential of each package.</p>
<p>Here’s your roadmap to mastering each component:</p>
<ul>
<li>
<strong><a href="https://glycoverse.github.io/glyexp/articles/glyexp.html" class="external-link">glyexp</a></strong>
— Master experiment objects and data manipulation</li>
<li>
<strong><a href="https://glycoverse.github.io/glyread/articles/glyread.html" class="external-link">glyread</a></strong>
— Import and organize glycoproteomics data</li>
<li>
<strong><a href="https://glycoverse.github.io/glyclean/articles/glyclean.html" class="external-link">glyclean</a></strong>
— Build custom preprocessing pipelines<br>
</li>
<li>
<strong><a href="https://glycoverse.github.io/glystats/articles/glystats.html" class="external-link">glystats</a></strong>
— Explore the full statistical toolkit</li>
<li>
<strong><a href="https://glycoverse.github.io/glyvis/articles/glyvis.html" class="external-link">glyvis</a></strong>
— Create stunning visualizations</li>
<li>
<strong><a href="https://glycoverse.github.io/glymotif/articles/glymotif.html" class="external-link">glymotif</a></strong>
— Define and analyze custom motifs</li>
<li>
<strong><a href="https://glycoverse.github.io/glydet/articles/glydet.html" class="external-link">glydet</a></strong>
— Create powerful derived traits</li>
<li>
<strong><a href="https://glycoverse.github.io/glyenzy/articles/glyenzy.html" class="external-link">glyenzy</a></strong>
— Explore enzyme-substrate relationships (we didn’t cover this one, but
it’s fascinating!)</li>
<li>
<strong><a href="https://glycoverse.github.io/glyrepr/articles/glyrepr.html" class="external-link">glyrepr</a></strong>
— Master glycan structure representation</li>
<li>
<strong><a href="https://glycoverse.github.io/glyparse/articles/glyparse.html" class="external-link">glyparse</a></strong>
— Parse and convert structural formats</li>
<li>
<strong><a href="https://glycoverse.github.io/glydraw/articles/glydraw.html" class="external-link">glydraw</a></strong>
— Draw glycan structures</li>
<li>
<strong><a href="https://glycoverse.github.io/glydb/articles/glydb.html" class="external-link">glydb</a></strong>
— Access glycan databases</li>
<li>
<strong><a href="https://glycoverse.github.io/glyanno/articles/glyanno.html" class="external-link">glyanno</a></strong>
— Annotate glycan structures</li>
<li>
<strong><a href="https://glycoverse.github.io/glysmith/articles/glysmith.html" class="external-link">glysmith</a></strong>
— Master the full analytical pipeline</li>
</ul>
<p>Happy glycan hunting! 🧬</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bin Fu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.2.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
