<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Case Study: Glycomics • glycoverse</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Case Study: Glycomics">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">glycoverse</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.0.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/case-study-1.html">Case Study: Glycoproteomics</a></li>
    <li><a class="dropdown-item" href="../articles/case-study-2.html">Case Study: Glycomics</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Case Study: Glycomics</h1>
            
      

      <div class="d-none name"><code>case-study-2.Rmd</code></div>
    </div>

    
    
<p>This vignette walks you through a complete glycomics analysis using
<code>glycoverse</code>. We’ll explore the full spectrum of glycomics
data analysis, from data loading and preprocessing to statistical
analysis and visualization. We’ll also dive into advanced glycan
structure analysis, including motif quantification and derived trait
analysis. Ready to dive in? Let’s go!</p>
<p><strong>Heads up:</strong> <code>glycoverse</code> is built on
<code>tidy</code> principles throughout. If you’re new to
<code>tidyverse</code> data analysis, we highly recommend checking out
Hadley Wickham’s excellent <a href="https://r4ds.hadley.nz" class="external-link">R for Data
Science</a>. Trust us, it’s worth the investment!</p>
<p>Quick readiness check:</p>
<ul>
<li>What’s a <code>tibble</code>?</li>
<li>How do you filter rows in a <code>tibble</code>?</li>
<li>What’s the modern alternative to <code>for</code> loops?</li>
<li>What’s the <code>%&gt;%</code> operator? Do we still need it?</li>
<li>What makes data “tidy”?</li>
</ul>
<div class="section level2">
<h2 id="loading-the-packages">Loading the Packages<a class="anchor" aria-label="anchor" href="#loading-the-packages"></a>
</h2>
<p>We first load the <code>tidyverse</code> package, as usual.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyverse.tidyverse.org" class="external-link">tidyverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Attaching core tidyverse packages</span> ──────────────────────── tidyverse 2.0.0 ──</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">dplyr    </span> 1.1.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">readr    </span> 2.1.5</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">forcats  </span> 1.0.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">stringr  </span> 1.5.2</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">ggplot2  </span> 4.0.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tibble   </span> 3.3.0</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">lubridate</span> 1.9.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">tidyr    </span> 1.3.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">purrr    </span> 1.1.0     </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ────────────────────────────────────────── tidyverse_conflicts() ──</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span> masks <span style="color: #0000BB;">stats</span>::filter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>    masks <span style="color: #0000BB;">stats</span>::lag()</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use the conflicted package (<span style="color: #0000BB; font-style: italic;">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors</span></span></code></pre></div>
<p>Just like <code>tidyverse</code>, <code>glycoverse</code> is a
meta-package that loads a collection of specialized packages all at
once.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://glycoverse.github.io/glycoverse/">glycoverse</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Attaching core glycoverse packages</span> ───────────────── glycoverse 0.0.0.<span style="color: #BB0000;">9000</span> ──</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyclean</span> 0.6.4     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyparse</span> 0.4.4</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glydet  </span> 0.2.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyread </span> 0.6.1</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyenzy </span> 0.2.2     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyrepr </span> 0.7.2</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyexp  </span> 0.9.1     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glystats</span> 0.4.2</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glymotif</span> 0.7.0     <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">glyvis  </span> 0.1.2</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Conflicts</span> ───────────────────────────────────────── glycoverse_conflicts() ──</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">glyclean</span>::<span style="color: #00BB00;">aggregate()</span>  masks <span style="color: #0000BB;">stats</span>::aggregate()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">filter()</span>        masks <span style="color: #0000BB;">stats</span>::filter()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">intersect()</span> masks <span style="color: #0000BB;">dplyr</span>::intersect(), <span style="color: #0000BB;">base</span>::intersect()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">lag()</span>           masks <span style="color: #0000BB;">stats</span>::lag()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">glyexp</span>::<span style="color: #00BB00;">select_var()</span>   masks <span style="color: #0000BB;">dplyr</span>::select_var()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">setdiff()</span>   masks <span style="color: #0000BB;">dplyr</span>::setdiff(), <span style="color: #0000BB;">base</span>::setdiff()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">dplyr</span>::<span style="color: #00BB00;">setequal()</span>      masks <span style="color: #0000BB;">base</span>::setequal()</span></span>
<span><span class="co">#&gt; <span style="color: #BB0000;">✖</span> <span style="color: #0000BB;">lubridate</span>::<span style="color: #00BB00;">union()</span>     masks <span style="color: #0000BB;">dplyr</span>::union(), <span style="color: #0000BB;">base</span>::union()</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Use the conflicted package (<span style="color: #0000BB; font-style: italic;">&lt;http://conflicted.r-lib.org/&gt;</span>) to force all conflicts to become errors</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="reading-the-data">Reading the Data<a class="anchor" aria-label="anchor" href="#reading-the-data"></a>
</h2>
<p>Data import is typically your first step in any analysis. For this
tutorial, we’ll use the <code>real_experiment2</code> dataset that comes
with <code>glyexp</code>. This is a real-world N-glycomics dataset from
144 patients across four liver conditions: healthy controls (H),
hepatitis (M), cirrhosis (Y), and hepatocellular carcinoma (C).</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">real_experiment2</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Experiment</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Expression matrix: <span style="color: #0000BB;">144</span> samples, <span style="color: #0000BB;">67</span> variables</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample information fields: <span style="color: #00BB00;">group</span> <span style="color: #0000BB;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Variable information fields: <span style="color: #00BB00;">glycan_composition</span> <span style="color: #0000BB;">&lt;comp&gt;</span>, <span style="color: #00BB00;">glycan_structure</span> <span style="color: #0000BB;">&lt;struct&gt;</span></span></span></code></pre></div>
<p>For your own glycomics projects, you can create an
<code>experiment()</code> object manually. See <a href="https://glycoverse.github.io/glyexp/articles/glyexp.html#building-your-own-data-empire" class="external-link">this
document</a> for more details.</p>
<p>The <code>real_experiment2</code> object is an
<code>experiment()</code> object. If you’ve worked with
<code>SummarizedExperiment</code> from Bioconductor, think of
<code>experiment()</code> as its tidy cousin. Essentially, it’s a smart
data container that manages three key components:</p>
<ul>
<li>
<strong>Expression matrix</strong>: quantitative data with samples
as columns and variables as rows</li>
<li>
<strong>Sample information</strong>: a tibble with sample metadata
(group, batch, demographics, etc.)</li>
<li>
<strong>Variable information</strong>: a tibble with feature
metadata (glycan compositions, glycan structures, etc.)</li>
</ul>
<p>You can get these data components by using
<code>get_expr_mat()</code>, <code>get_sample_info()</code>, and
<code>get_var_info()</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_expr_mat</span><span class="op">(</span><span class="va">real_experiment2</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;          S1       S2       S3       S4       S5</span></span>
<span><span class="co">#&gt; V1 1354.352 1884.387 1389.444 2034.693 1472.504</span></span>
<span><span class="co">#&gt; V2 3315.779 2500.308 1247.036 3102.668 2903.602</span></span>
<span><span class="co">#&gt; V3 6940.940 5911.016 1686.319 4071.061 4349.991</span></span>
<span><span class="co">#&gt; V4 4437.816 7535.886 2053.077 4773.249 3142.817</span></span>
<span><span class="co">#&gt; V5 1346.274 1663.375 1043.464 1765.550 1184.083</span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_sample_info</span><span class="op">(</span><span class="va">real_experiment2</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 144 × 2</span></span></span>
<span><span class="co">#&gt;    sample group</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> S1     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> S2     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> S3     Y    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> S4     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> S5     H    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> S6     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> S7     M    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> S8     C    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> S9     M    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> S10    M    </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 134 more rows</span></span></span></code></pre></div>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_var_info</span><span class="op">(</span><span class="va">real_experiment2</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 67 × 3</span></span></span>
<span><span class="co">#&gt;    variable glycan_composition                  </span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;comp&gt;</span>                              </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V1       Man(3)GlcNAc(3)                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V2       Man(3)GlcNAc(7)                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V3       Man(5)GlcNAc(2)                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V4       Man(4)Gal(2)GlcNAc(4)Neu5Ac(2)      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V5       Man(3)Gal(1)GlcNAc(3)               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V6       Man(3)Gal(2)GlcNAc(4)Fuc(2)         </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V7       Man(3)GlcNAc(3)Fuc(1)               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V8       Man(3)GlcNAc(4)                     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V9       Man(3)Gal(2)GlcNAc(5)Neu5Ac(1)      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V10      Man(3)Gal(1)GlcNAc(5)Fuc(1)Neu5Ac(1)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 57 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: glycan_structure &lt;struct&gt;</span></span></span></code></pre></div>
<p>What makes <code>experiment()</code> objects so powerful?</p>
<ol style="list-style-type: decimal">
<li><p><strong>Universal compatibility</strong>: All
<code>glycoverse</code> functions speak the same language. They
automatically know where to find what they need in an
<code>experiment()</code> object, which eliminates the biggest headache
in data analysis: constantly reformatting data to meet different
function requirements.</p></li>
<li><p><strong>Synchronized operations</strong>: You can use familiar
<code>dplyr</code>-style functions to filter, select, and modify your
data, and everything stays in sync automatically. No more worrying about
mismatched sample names or lost metadata!</p></li>
</ol>
<p>We’ll see these benefits in action throughout this tutorial. For a
deeper dive into <code>experiment()</code> objects, check out <a href="https://glycoverse.github.io/glyexp/articles/glyexp.html" class="external-link">Get
Started with glyexp</a>.</p>
</div>
<div class="section level2">
<h2 id="data-preprocessing">Data Preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h2>
<p>Raw quantification data needs preprocessing before analysis—that’s
just a fact of life in omics. Typical steps include normalization,
missing value imputation, and batch effect correction. Rather than
making you implement these tedious steps manually, <code>glyclean</code>
provides a comprehensive preprocessing pipeline. Just call
<code>auto_clean()</code> on your <code>experiment()</code> object and
you’re good to go.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clean_exp</span> <span class="op">&lt;-</span> <span class="fu">auto_clean</span><span class="op">(</span><span class="va">real_experiment2</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing data (Median Quotient)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Normalizing data (Median Quotient) <span style="color: #B2B2B2;">[18ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Removing variables with &gt;50% missing values</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Removing variables with &gt;50% missing values <span style="color: #B2B2B2;">[28ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Imputing missing values</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample size &gt; 100, using MissForest imputation</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Imputing missing values<span style="color: #00BB00;">✔</span> Imputing missing values <span style="color: #B2B2B2;">[14.2s]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Normalizing data (Total Area)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Normalizing data (Total Area) <span style="color: #B2B2B2;">[14ms]</span></span></span></code></pre></div>
<p>Your data is now analysis-ready!</p>
<p>Want to customize the preprocessing steps? See <a href="https://glycoverse.github.io/glyclean/articles/glyclean.html" class="external-link">Get
Started with glyclean</a> for the full toolkit.</p>
</div>
<div class="section level2">
<h2 id="statistical-analysis-and-visualization">Statistical Analysis and Visualization<a class="anchor" aria-label="anchor" href="#statistical-analysis-and-visualization"></a>
</h2>
<p>Time for the fun part—statistical analysis and visualization! We’ll
use <code>glystats</code> for the number crunching and
<code>glyvis</code> to make sense of the results visually.</p>
<p>Let’s kick off with PCA to get a bird’s-eye view of our data
structure.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">plot_pca</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glyvis`</span></span></code></pre></div>
<p><img src="case-study-2_files/figure-html/unnamed-chunk-9-1.png" width="384"></p>
<p><code>glyvis</code> isn’t designed for publication-ready figures, but
it’s perfect for quick exploratory visualization. Behind the scenes,
<code>plot_pca()</code> calls <code>gly_pca()</code> from
<code>glystats</code> and renders the results.</p>
<p>You can also break this down into separate steps:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca_res</span> <span class="op">&lt;-</span> <span class="fu">gly_pca</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glystats`</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot</a></span><span class="op">(</span><span class="va">pca_res</span><span class="op">)</span>  <span class="co"># from `glyvis`</span></span></code></pre></div>
<p><img src="case-study-2_files/figure-html/unnamed-chunk-10-1.png" width="384"></p>
<p>We actually recommend the two-step approach, since it gives you more
flexibility with the results. You can create custom <code>ggplot2</code>
visualizations for publications or extract the underlying data when
reviewers ask for it.</p>
<p><code>glystats</code> covers virtually all standard omics analyses.
All functions follow the same naming pattern:
<code>gly_xxx()</code>—think <code>gly_anova()</code>,
<code>gly_ttest()</code>, <code>gly_roc()</code>,
<code>gly_cox()</code>, <code>gly_wgcna()</code>, and so on. They all
take an <code>experiment()</code> object as their first argument.</p>
<p>The return format is consistent across all functions—a list with two
components:</p>
<ul>
<li>
<strong><code>tidy_result</code></strong>: cleaned-up tibbles in
tidy format. We’ve done the heavy lifting of organizing messy
statistical output for you.</li>
<li>
<strong><code>raw_result</code></strong>: the original statistical
objects. These are available when you need to dig deeper or perform
advanced analyses.</li>
</ul>
<p>Let’s peek inside the <code>pca_res</code> object we just
created:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/str.html" class="external-link">str</a></span><span class="op">(</span><span class="va">pca_res</span>, max.level <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">#&gt; List of 2</span></span>
<span><span class="co">#&gt;  $ tidy_result:List of 3</span></span>
<span><span class="co">#&gt;   ..$ samples    : tibble [7,488 × 4] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ variables  : tibble [2,704 × 5] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;   ..$ eigenvalues: tibble [52 × 4] (S3: tbl_df/tbl/data.frame)</span></span>
<span><span class="co">#&gt;  $ raw_result :List of 5</span></span>
<span><span class="co">#&gt;   ..$ sdev    : num [1:52] 4.22 2.52 2.37 1.85 1.57 ...</span></span>
<span><span class="co">#&gt;   ..$ rotation: num [1:52, 1:52] -0.0755 -0.0905 -0.1817 -0.0532 -0.1163 ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..$ center  : Named num [1:52] 0.000984 0.001836 0.003949 0.003128 0.000923 ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "names")= chr [1:52] "V1" "V2" "V3" "V4" ...</span></span>
<span><span class="co">#&gt;   ..$ scale   : Named num [1:52] 0.000449 0.000553 0.001532 0.002131 0.000291 ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "names")= chr [1:52] "V1" "V2" "V3" "V4" ...</span></span>
<span><span class="co">#&gt;   ..$ x       : num [1:144, 1:52] 1.168 3.486 -0.165 2.036 5.837 ...</span></span>
<span><span class="co">#&gt;   .. ..- attr(*, "dimnames")=List of 2</span></span>
<span><span class="co">#&gt;   ..- attr(*, "class")= chr "prcomp"</span></span>
<span><span class="co">#&gt;  - attr(*, "class")= chr [1:2] "glystats_pca_res" "glystats_res"</span></span></code></pre></div>
<p>As you can see, <code>pca_res$tidy_result</code> contains three tidy
tibbles, while <code>pca_res$raw_result</code> holds the original
<code>prcomp</code> object from <code><a href="https://rdrr.io/r/stats/prcomp.html" class="external-link">stats::prcomp()</a></code>. Check
<code>?gly_pca</code> for the complete breakdown.</p>
<p>Here’s what the <code>samples</code> tibble looks like:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pca_res</span><span class="op">$</span><span class="va">tidy_result</span><span class="op">$</span><span class="va">samples</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7,488 × 4</span></span></span>
<span><span class="co">#&gt;    sample group    PC  value</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> S1     H         1  1.17 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> S1     H         2  1.17 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> S1     H         3 -<span style="color: #BB0000;">1.18</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> S1     H         4 -<span style="color: #BB0000;">1.79</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> S1     H         5  1.29 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> S1     H         6 -<span style="color: #BB0000;">1.89</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> S1     H         7 -<span style="color: #BB0000;">2.23</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> S1     H         8  1.48 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> S1     H         9 -<span style="color: #BB0000;">1.65</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> S1     H        10 -<span style="color: #BB0000;">0.477</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 7,478 more rows</span></span></span></code></pre></div>
<p>Notice the “group” column? That’s <code>glystats</code> being
helpful— it automatically pulls relevant metadata from your
<code>experiment()</code> object and includes it in the results wherever
it makes sense.</p>
<p>Back to that <code><a href="https://ggplot2.tidyverse.org/reference/autoplot.html" class="external-link">autoplot()</a></code> magic we saw earlier. It
automatically recognizes different <code>glystats</code> result types
and plots accordingly— no manual specification needed. The plots won’t
win any beauty contests, but they’ll get your data insights across
fast.</p>
<p>Now let’s dive into differential expression analysis using the
tried-and-true <code>limma</code> package.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_res</span> <span class="op">&lt;-</span> <span class="fu">gly_limma</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glystats`</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of groups: <span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Groups: <span style="color: #0000BB;">"C"</span>, <span style="color: #0000BB;">"H"</span>, <span style="color: #0000BB;">"M"</span>, and <span style="color: #0000BB;">"Y"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Performing multi-group limma analysis with 4 groups</span></span>
<span><span class="va">limma_res</span><span class="op">$</span><span class="va">tidy_result</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 312 × 10</span></span></span>
<span><span class="co">#&gt;    variable glycan_composition</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;comp&gt;</span>            </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V1       Man(3)GlcNAc(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V1       Man(3)GlcNAc(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V1       Man(3)GlcNAc(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V1       Man(3)GlcNAc(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V1       Man(3)GlcNAc(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V1       Man(3)GlcNAc(3)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V2       Man(3)GlcNAc(7)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V2       Man(3)GlcNAc(7)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V2       Man(3)GlcNAc(7)   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V2       Man(3)GlcNAc(7)   </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 302 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 8 more variables: glycan_structure &lt;struct&gt;, log2fc &lt;dbl&gt;, AveExpr &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   t &lt;dbl&gt;, p &lt;dbl&gt;, p_adj &lt;dbl&gt;, b &lt;dbl&gt;, contrast &lt;chr&gt;</span></span></span></code></pre></div>
<p>Excellent! Now let’s identify significantly differentially expressed
glycans between HCC and healthy samples.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">limma_res</span><span class="op">$</span><span class="va">tidy_result</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">contrast</span> <span class="op">==</span> <span class="st">"C_vs_H"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">glycan_composition</span>, <span class="va">p_adj</span>, <span class="va">log2fc</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 15 × 3</span></span></span>
<span><span class="co">#&gt;    glycan_composition                          p_adj    log2fc</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;comp&gt;</span>                                      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Man(3)GlcNAc(4)Fuc(1)                0.000<span style="text-decoration: underline;">000</span>449   0.017<span style="text-decoration: underline;">2</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Man(3)GlcNAc(5)                      0.028<span style="text-decoration: underline;">2</span>        0.000<span style="text-decoration: underline;">768</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> Man(3)Gal(1)GlcNAc(5)Neu5Ac(1)       0.028<span style="text-decoration: underline;">5</span>        0.001<span style="text-decoration: underline;">05</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Man(3)GlcNAc(5)Fuc(1)                0.000<span style="text-decoration: underline;">411</span>      0.002<span style="text-decoration: underline;">91</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Man(3)Gal(1)GlcNAc(5)Fuc(1)          0.004<span style="text-decoration: underline;">41</span>       0.004<span style="text-decoration: underline;">93</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Man(3)Gal(2)GlcNAc(4)Fuc(1)Neu5Ac(2) 0.041<span style="text-decoration: underline;">8</span>        0.011<span style="text-decoration: underline;">4</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Man(3)Gal(1)GlcNAc(5)                0.019<span style="text-decoration: underline;">5</span>        0.001<span style="text-decoration: underline;">14</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> Man(3)Gal(1)GlcNAc(4)Fuc(1)          0.017<span style="text-decoration: underline;">3</span>        0.013<span style="text-decoration: underline;">7</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Man(3)Gal(2)GlcNAc(4)Neu5Ac(1)       0.003<span style="text-decoration: underline;">34</span>      -<span style="color: #BB0000;">0.014</span><span style="color: #BB0000; text-decoration: underline;">1</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Man(3)Gal(3)GlcNAc(5)Neu5Ac(2)       0.000<span style="text-decoration: underline;">053</span>5    -<span style="color: #BB0000;">0.004</span><span style="color: #BB0000; text-decoration: underline;">77</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> Man(3)Gal(3)GlcNAc(5)Fuc(1)Neu5Ac(2) 0.000<span style="text-decoration: underline;">000</span>449   0.004<span style="text-decoration: underline;">06</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> Man(3)Gal(3)GlcNAc(5)Neu5Ac(3)       0.001<span style="text-decoration: underline;">06</span>      -<span style="color: #BB0000;">0.026</span><span style="color: #BB0000; text-decoration: underline;">4</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> Man(3)Gal(4)GlcNAc(6)Neu5Ac(2)       0.013<span style="text-decoration: underline;">0</span>        0.001<span style="text-decoration: underline;">28</span> </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> Man(3)Gal(3)GlcNAc(5)Fuc(1)Neu5Ac(3) 0.000<span style="text-decoration: underline;">000</span>023<span style="text-decoration: underline;">7</span>  0.034<span style="text-decoration: underline;">2</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> Man(3)Gal(3)GlcNAc(5)Fuc(2)Neu5Ac(3) 0.000<span style="text-decoration: underline;">411</span>      0.001<span style="text-decoration: underline;">97</span></span></span></code></pre></div>
<p>For the full statistical arsenal, check out <a href="https://glycoverse.github.io/glystats/articles/glystats.html" class="external-link">Get
Started with glystats</a> and <a href="https://glycoverse.github.io/glyvis/articles/glyvis.html" class="external-link">Get
Started with glyvis</a>.</p>
</div>
<div class="section level2">
<h2 id="understanding-your-data-better">Understanding Your Data Better<a class="anchor" aria-label="anchor" href="#understanding-your-data-better"></a>
</h2>
<p>We want to subset the experiment to keep only “H” or “C” samples, and
only the significant glycans above.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_vars</span> <span class="op">&lt;-</span> <span class="va">limma_res</span><span class="op">$</span><span class="va">tidy_result</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">contrast</span> <span class="op">==</span> <span class="st">"C_vs_H"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sub_exp</span> <span class="op">&lt;-</span> <span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_var</span><span class="op">(</span><span class="va">variable</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">sig_vars</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_obs</span><span class="op">(</span><span class="va">group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"H"</span>, <span class="st">"C"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sub_exp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Experiment</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Expression matrix: <span style="color: #0000BB;">73</span> samples, <span style="color: #0000BB;">15</span> variables</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample information fields: <span style="color: #00BB00;">group</span> <span style="color: #0000BB;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Variable information fields: <span style="color: #00BB00;">glycan_composition</span> <span style="color: #0000BB;">&lt;comp&gt;</span>, <span style="color: #00BB00;">glycan_structure</span> <span style="color: #0000BB;">&lt;struct&gt;</span></span></span></code></pre></div>
<p>That last code chunk showcased some powerful <code>glyexp</code>
features. Let’s take a moment to understand what’s really happening
under the hood.</p>
<p>First, what exactly are those <code>sig_vars</code>? At first glance,
they look pretty cryptic:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_vars</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt; [1] "V19" "V20" "V21" "V27" "V32"</span></span></code></pre></div>
<p>Let’s revisit the variable information tibble:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_var_info</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 5 × 3</span></span></span>
<span><span class="co">#&gt;   variable glycan_composition            </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;comp&gt;</span>                        </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> V1       Man(3)GlcNAc(3)               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> V2       Man(3)GlcNAc(7)               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> V3       Man(5)GlcNAc(2)               </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> V4       Man(4)Gal(2)GlcNAc(4)Neu5Ac(2)</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> V5       Man(3)Gal(1)GlcNAc(3)         </span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 1 more variable: glycan_structure &lt;struct&gt;</span></span></span></code></pre></div>
<p>See that first “variable” column? That’s the index column that keeps
everything synchronized. It links the variable information tibble to the
expression matrix. Take a look at the expression matrix row names:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_expr_mat</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span></span>
<span><span class="co">#&gt;              S1           S2          S3           S4           S5</span></span>
<span><span class="co">#&gt; V1 0.0008735189 0.0010297263 0.001797452 0.0010678250 0.0008232803</span></span>
<span><span class="co">#&gt; V2 0.0021385840 0.0013662972 0.001613227 0.0016283078 0.0016234107</span></span>
<span><span class="co">#&gt; V3 0.0044767113 0.0032300837 0.002181504 0.0021365292 0.0024320901</span></span>
<span><span class="co">#&gt; V4 0.0028622668 0.0041179968 0.002655960 0.0025050435 0.0017571563</span></span>
<span><span class="co">#&gt; V5 0.0008683088 0.0009089539 0.001349876 0.0009265763 0.0006620237</span></span></code></pre></div>
<p>Notice how the row names match the “variable” column exactly, in the
same order? The same principle applies to samples—the “sample” column in
sample info matches the expression matrix column names. This is how
<code>experiment()</code> keeps all three data components perfectly
synchronized.</p>
<p><img src="experiment.png"></p>
<p>Of course, synchronization only matters when you’re actually
manipulating data. <code>glyexp</code> provides a full suite of
<code>dplyr</code>-style functions for data manipulation. They work just
like regular <code>dplyr</code> functions, but with helpful suffixes:
<code>xxx_obs()</code> functions operate on samples (observations), and
<code>xxx_var()</code> functions work with variables (features).</p>
<p>For instance, let’s filter to just the healthy samples using
<code>filter_obs()</code>:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Filter the data</span></span>
<span><span class="va">exp_h</span> <span class="op">&lt;-</span> <span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_obs</span><span class="op">(</span><span class="va">group</span> <span class="op">==</span> <span class="st">"H"</span><span class="op">)</span></span></code></pre></div>
<p>The syntax should feel familiar if you know
<code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">dplyr::filter()</a></code>. The magic happens behind the scenes—the
expression matrix updates automatically. Let’s verify the filtering
worked by checking the dimensions:</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Before filtering</span></span>
<span><span class="fu">cli</span><span class="fu">::</span><span class="fu"><a href="https://cli.r-lib.org/reference/cli_text.html" class="external-link">cli_text</a></span><span class="op">(</span><span class="st">"Number of rows in sample information tibble: {nrow(get_sample_info(clean_exp))}"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of rows in sample information tibble: 144</span></span>
<span><span class="fu">cli</span><span class="fu">::</span><span class="fu"><a href="https://cli.r-lib.org/reference/cli_text.html" class="external-link">cli_text</a></span><span class="op">(</span><span class="st">"Number of rows in variable information tibble: {nrow(get_var_info(clean_exp))}"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of rows in variable information tibble: 52</span></span>
<span><span class="fu">cli</span><span class="fu">::</span><span class="fu"><a href="https://cli.r-lib.org/reference/cli_text.html" class="external-link">cli_text</a></span><span class="op">(</span><span class="st">"Dimensions of expression matrix: {ncol(get_expr_mat(clean_exp))} x {nrow(get_expr_mat(clean_exp))}"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Dimensions of expression matrix: 144 x 52</span></span>
<span></span>
<span><span class="co"># After filtering</span></span>
<span><span class="fu">cli</span><span class="fu">::</span><span class="fu"><a href="https://cli.r-lib.org/reference/cli_text.html" class="external-link">cli_text</a></span><span class="op">(</span><span class="st">"Number of rows in sample information tibble: {nrow(get_sample_info(exp_h))}"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of rows in sample information tibble: 26</span></span>
<span><span class="fu">cli</span><span class="fu">::</span><span class="fu"><a href="https://cli.r-lib.org/reference/cli_text.html" class="external-link">cli_text</a></span><span class="op">(</span><span class="st">"Number of rows in variable information tibble: {nrow(get_var_info(exp_h))}"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Number of rows in variable information tibble: 52</span></span>
<span><span class="fu">cli</span><span class="fu">::</span><span class="fu"><a href="https://cli.r-lib.org/reference/cli_text.html" class="external-link">cli_text</a></span><span class="op">(</span><span class="st">"Dimensions of expression matrix: {ncol(get_expr_mat(exp_h))} x {nrow(get_expr_mat(exp_h))}"</span><span class="op">)</span></span>
<span><span class="co">#&gt; Dimensions of expression matrix: 26 x 52</span></span></code></pre></div>
<p>In essence, <code>filter_obs()</code> applies
<code><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">dplyr::filter()</a></code> to the sample information and automatically
updates the expression matrix to match. <code>filter_var()</code> does
the same for variables. For the complete catalog of
<code>dplyr</code>-style functions, check out <a href="https://glycoverse.github.io/glyexp/articles/dplyr-style-functions.html" class="external-link">this
vignette</a>.</p>
<p>Now that code snippet should make perfect sense:</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sig_vars</span> <span class="op">&lt;-</span> <span class="va">limma_res</span><span class="op">$</span><span class="va">tidy_result</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span>, <span class="va">contrast</span> <span class="op">==</span> <span class="st">"C_vs_H"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">variable</span><span class="op">)</span></span>
<span></span>
<span><span class="va">sub_exp</span> <span class="op">&lt;-</span> <span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_var</span><span class="op">(</span><span class="va">variable</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">sig_vars</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">filter_obs</span><span class="op">(</span><span class="va">group</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"H"</span>, <span class="st">"C"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="advanced-motif-analysis">Advanced Motif Analysis<a class="anchor" aria-label="anchor" href="#advanced-motif-analysis"></a>
</h2>
<p>Up to now, we’ve covered standard glycomics workflows. While
<code>glycoverse</code> certainly streamlines these analyses, it truly
shines when it comes to advanced glycan structure analysis.</p>
<p>Before diving into motifs, let’s get acquainted with
<code><a href="https://glycoverse.github.io/glyrepr/reference/glycan_structure.html" class="external-link">glyrepr::glycan_structure()</a></code> vectors.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clean_exp</span><span class="op">$</span><span class="va">var_info</span><span class="op">[[</span><span class="st">"glycan_structure"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="co">#&gt; &lt;glycan_structure[52]&gt;</span></span>
<span><span class="co">#&gt; [1] <span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [2] <span style="color: #0087D7;">GlcNAc</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)][<span style="color: #0087D7;">GlcNAc</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [3] <span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [4] <span style="color: #AF5FAF;">Neu5Ac</span>(?2-?)<span style="color: #FFD700;">Gal</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)[<span style="color: #AF5FAF;">Neu5Ac</span>(?2-?)<span style="color: #FFD700;">Gal</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [5] <span style="color: #FFD700;">Gal</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [6] <span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)[<span style="color: #FF0000;">Fuc</span>(?1-?)]<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [7] <span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [8] <span style="color: #AF5FAF;">Neu5Ac</span>(?2-?)<span style="color: #FFD700;">Gal</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)][<span style="color: #FFD700;">Gal</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [9] <span style="color: #AF5FAF;">Neu5Ac</span>(?2-?)<span style="color: #FFD700;">Gal</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)][<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)[<span style="color: #FF0000;">Fuc</span>(?1-?)]<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; [10] <span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)[<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #00AF5F;">Man</span>(?1-?)]<span style="color: #00AF5F;">Man</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(?1-?)<span style="color: #0087D7;">GlcNAc</span>(<span style="color: #B2B2B2;">?1-</span></span></span>
<span><span class="co">#&gt; ... (42 more not shown)</span></span>
<span><span class="co">#&gt; # Unique structures: 52</span></span></code></pre></div>
<p>Just like <code><a href="https://rdrr.io/r/base/integer.html" class="external-link">integer()</a></code> and <code><a href="https://rdrr.io/r/base/character.html" class="external-link">character()</a></code>,
<code>glycan_structure()</code> is a specialized vector type. Some
software (like pGlyco3 and StrucGP) outputs structural information as
text strings. When you import this data using <code>glyread</code>, the
<code>glyparse</code> package automatically converts these strings into
proper <code>glycan_structure()</code> vectors and stores them in the
variable information tibble. Note that not all software provides
structural data—some only give compositions.</p>
<p>For glycomics data, this information is hard to come by
automatically. You can do it manually by parsing the glycan structure
strings using <code>glyparse</code> and using
<code>left_join_var()</code> to join the parsed structures to the
variable information tibble.</p>
<p>Fortunately, our example dataset includes structural information,
opening up a world of advanced analytical possibilities. Let’s explore
motif analysis.</p>
<p><strong>Quick note:</strong> The printed structures use
IUPAC-condensed notation, which we’ll also use for defining motifs
below. Don’t worry if it looks intimidating—we’ll include visual
diagrams to help. That said, if you’re planning to do serious structural
analysis, learning IUPAC-condensed notation is worth the investment.
Check out <a href="https://glycoverse.github.io/glyrepr/articles/iupac.html" class="external-link">this
guide</a> to get started—it’s easier than it looks!</p>
<p>Human serum N-glycans can have three types of branch terminals
(ignoring a1-3 Fuc):</p>
<ol style="list-style-type: decimal">
<li>A Sialyl-LacNAc motif</li>
<li>A LacNAc motif without sialic acids</li>
<li>Only a GlcNAc without further elongation</li>
</ol>
<p><img src="branch_motifs.png"></p>
<p>Here’s how we express these motifs in IUPAC-condensed notation:</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motifs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span></span>
<span>  motif1 <span class="op">=</span> <span class="st">"Neu5Ac(??-?)Gal(??-?)GlcNAc(??-"</span>,</span>
<span>  motif2 <span class="op">=</span> <span class="st">"Gal(??-?)GlcNAc(??-"</span>,</span>
<span>  motif3 <span class="op">=</span> <span class="st">"GlcNAc(??-"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>The “??-?” represents unknown linkages—a common limitation in mass
spectrometry data.</p>
<p>Here’s our research question: <strong>Which branching motif show
differential expression across conditions?</strong> Without
<code>glycoverse</code>, this would be a nightmare to tackle manually.
Take a moment to imagine the pain of doing this by hand!</p>
<p>Now, the <code>glycoverse</code> solution:</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motif_anova_res</span> <span class="op">&lt;-</span> <span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">quantify_motifs</span><span class="op">(</span><span class="va">motifs</span>, alignments <span class="op">=</span> <span class="st">"terminal"</span><span class="op">)</span> <span class="op">|&gt;</span>  <span class="co"># quantify these motifs</span></span>
<span>  <span class="fu">gly_anova</span><span class="op">(</span><span class="op">)</span>  <span class="co"># and perform ANOVA</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of groups: <span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Groups: <span style="color: #0000BB;">"C"</span>, <span style="color: #0000BB;">"H"</span>, <span style="color: #0000BB;">"M"</span>, and <span style="color: #0000BB;">"Y"</span></span></span>
<span></span>
<span><span class="va">motif_anova_res</span><span class="op">$</span><span class="va">tidy_result</span><span class="op">$</span><span class="va">main_test</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 10</span></span></span>
<span><span class="co">#&gt;   variable motif  term     df  sumsq  meansq statistic  p_value   p_adj post_hoc</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> V1       motif1 group     3 0.020<span style="text-decoration: underline;">9</span> 0.006<span style="text-decoration: underline;">95</span>      1.18 0.319    0.319   <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> V2       motif2 group     3 0.025<span style="text-decoration: underline;">7</span> 0.008<span style="text-decoration: underline;">56</span>      1.93 0.128    0.192   <span style="color: #BB0000;">NA</span>      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> V3       motif3 group     3 0.102  0.034<span style="text-decoration: underline;">1</span>       5.90 0.000<span style="text-decoration: underline;">808</span> 0.002<span style="text-decoration: underline;">42</span> H-C;M-C…</span></span></code></pre></div>
<p><code>quantify_motifs()</code> transforms your data into a new
<code>experiment()</code> object. Instead of quantification of glycans,
you now have motif abundances across samples. Since it’s still an
<code>experiment()</code> object, all <code>glystats</code> functions
work seamlessly—including <code>gly_anova()</code>.</p>
<p>Now we can answer our question using standard <code>tidyverse</code>
operations, since <code>motif_anova_res$tidy_result$main_test</code> is
just a regular tibble:</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">motif_anova_res</span><span class="op">$</span><span class="va">tidy_result</span><span class="op">$</span><span class="va">main_test</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 10</span></span></span>
<span><span class="co">#&gt;   variable motif  term     df sumsq meansq statistic  p_value   p_adj post_hoc  </span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> V3       motif3 group     3 0.102 0.034<span style="text-decoration: underline;">1</span>      5.90 0.000<span style="text-decoration: underline;">808</span> 0.002<span style="text-decoration: underline;">42</span> H-C;M-C;Y…</span></span></code></pre></div>
<p>Here’s another common question: <strong>Which of the three branching
motifs appears the most in all glycans?</strong></p>
<p>For this analysis, we don’t need motif quantification—we just need to
know which glycans have these motifs.
<code><a href="https://glycoverse.github.io/glymotif/reference/add_motifs_int.html" class="external-link">glymotif::add_motifs_lgl()</a></code> is perfect for this.</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clean_exp</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">add_motifs_lgl</span><span class="op">(</span><span class="va">motifs</span>, alignments <span class="op">=</span> <span class="st">"terminal"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu">get_var_info</span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">glycan_composition</span>, <span class="va">motif1</span>, <span class="va">motif2</span>, <span class="va">motif3</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span><span class="op">-</span><span class="va">glycan_composition</span>, names_to <span class="op">=</span> <span class="st">"motif"</span>, values_to <span class="op">=</span> <span class="st">"has_motif"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">has_motif</span><span class="op">)</span>, .by <span class="op">=</span> <span class="st">"motif"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 3 × 2</span></span></span>
<span><span class="co">#&gt;   motif      n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> motif1    27</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> motif2    21</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> motif3    23</span></span></code></pre></div>
<p><code>add_motifs_lgl()</code> adds three new TRUE/FALSE columns
(<code>motif1</code>, <code>motif2</code>, <code>motif3</code>) to the
variable information.</p>
<p><code>glymotif</code> has much more to offer beyond these examples.
Dive deeper with <a href="https://glycoverse.github.io/glymotif/articles/glymotif.html" class="external-link">Get
Started with glymotif</a>.</p>
</div>
<div class="section level2">
<h2 id="derived-trait-analysis">Derived Trait Analysis<a class="anchor" aria-label="anchor" href="#derived-trait-analysis"></a>
</h2>
<p>Let’s wrap up with derived traits—a clever analytical approach
developed by the N-glycomics community for glycome characterization.
Classic examples include:</p>
<ul>
<li>High-mannose glycan proportion</li>
<li>Core-fucosylation rate within complex glycans</li>
<li>Average sialylation per galactose residue</li>
</ul>
<p><code>glydet</code> calculates derived traits in a flash. Using it
couldn’t be simpler:</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trait_exp</span> <span class="op">&lt;-</span> <span class="fu">derive_traits</span><span class="op">(</span><span class="va">clean_exp</span><span class="op">)</span>  <span class="co"># from `glydet`</span></span>
<span><span class="va">trait_exp</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">──</span> <span style="font-weight: bold;">Experiment</span> <span style="color: #00BBBB;">──────────────────────────────────────────────────────────────────</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Expression matrix: <span style="color: #0000BB;">144</span> samples, <span style="color: #0000BB;">14</span> variables</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Sample information fields: <span style="color: #00BB00;">group</span> <span style="color: #0000BB;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Variable information fields: <span style="color: #00BB00;">trait</span> <span style="color: #0000BB;">&lt;chr&gt;</span></span></span></code></pre></div>
<p>That’s it! Just like <code>quantify_motifs()</code>,
<code>derive_traits()</code> creates a new <code>experiment()</code>
object, but now with trait values per sample.</p>
<p>The variable information shows what we’re working with:</p>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">get_var_info</span><span class="op">(</span><span class="va">trait_exp</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 14 × 2</span></span></span>
<span><span class="co">#&gt;    variable trait</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> V1       TM   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> V2       TH   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> V3       TC   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> V4       MM   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> V5       CA2  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> V6       CA3  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> V7       CA4  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> V8       TF   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> V9       TFc  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> V10      TFa  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> V11      TB   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> V12      SG   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> V13      GA   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> V14      TS</span></span></code></pre></div>
<p>The “trait” column lists all the derived traits we can analyze.</p>
<p><code>glydet</code> comes with a comprehensive set of built-in
traits:</p>
<ul>
<li>
<strong><code>TM</code></strong>: Proportion of high-mannose
glycans</li>
<li>
<strong><code>TH</code></strong>: Proportion of hybrid glycans<br>
</li>
<li>
<strong><code>TC</code></strong>: Proportion of complex glycans</li>
<li>
<strong><code>MM</code></strong>: Average number of mannoses within
high-mannose glycans</li>
<li>
<strong><code>CA2</code></strong>: Proportion of bi-antennary
glycans within complex glycans</li>
<li>
<strong><code>CA3</code></strong>: Proportion of tri-antennary
glycans within complex glycans</li>
<li>
<strong><code>CA4</code></strong>: Proportion of tetra-antennary
glycans within complex glycans</li>
<li>
<strong><code>TF</code></strong>: Proportion of fucosylated
glycans</li>
<li>
<strong><code>TFc</code></strong>: Proportion of core-fucosylated
glycans</li>
<li>
<strong><code>TFa</code></strong>: Proportion of arm-fucosylated
glycans</li>
<li>
<strong><code>TB</code></strong>: Proportion of glycans with
bisecting GlcNAc</li>
<li>
<strong><code>SG</code></strong>: Average degree of sialylation per
galactose</li>
<li>
<strong><code>GA</code></strong>: Average degree of galactosylation
per antenna</li>
<li>
<strong><code>TS</code></strong>: Proportion of sialylated
glycans</li>
</ul>
<p>These represent the most widely used traits in glycomics
literature.</p>
<p>Let’s identify traits with significantly different values across
conditions:</p>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">trait_anova_res</span> <span class="op">&lt;-</span> <span class="fu">gly_anova</span><span class="op">(</span><span class="va">trait_exp</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Number of groups: <span style="color: #0000BB;">4</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Groups: <span style="color: #0000BB;">"C"</span>, <span style="color: #0000BB;">"H"</span>, <span style="color: #0000BB;">"M"</span>, and <span style="color: #0000BB;">"Y"</span></span></span>
<span><span class="va">trait_anova_res</span><span class="op">$</span><span class="va">tidy_result</span><span class="op">$</span><span class="va">main_test</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">p_adj</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 8 × 10</span></span></span>
<span><span class="co">#&gt;   variable trait term     df    sumsq  meansq statistic p_value   p_adj post_hoc</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> V5       CA2   group     3  8.60<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.87<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>      5.70 1.04<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> M-C;Y-C </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> V6       CA3   group     3  2.19<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 7.30<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>      5.63 1.13<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 2.65<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> M-C;Y-C </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> V7       CA4   group     3  6.94<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 2.31<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span>      6.08 6.42<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 2.25<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> M-C     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> V8       TF    group     3  1.08<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>      7.88 6.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 3.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> H-C;M-C…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> V9       TFc   group     3  1.08<span style="color: #949494;">e</span><span style="color: #BB0000;">-1</span> 3.62<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span>      7.88 6.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 3.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> H-C;M-C…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> V10      TFa   group     3  1.31<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> 4.35<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span>      7.85 6.99<span style="color: #949494;">e</span><span style="color: #BB0000;">-5</span> 3.26<span style="color: #949494;">e</span><span style="color: #BB0000;">-4</span> H-C;M-C…</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> V11      TB    group     3  1.14<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 3.81<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>      3.59 1.55<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 2.71<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> Y-H     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">8</span> V13      GA    group     3  5.35<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span> 1.78<span style="color: #949494;">e</span><span style="color: #BB0000;">-3</span>      3.85 1.10<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> 2.20<span style="color: #949494;">e</span><span style="color: #BB0000;">-2</span> H-C</span></span></code></pre></div>
<p>Once again, it’s just that straightforward.</p>
<p>This just scratches the surface of <code>glydet</code>’s
capabilities. The real power lies in defining custom traits tailored to
your research questions. Explore the possibilities in <a href="https://glycoverse.github.io/glydet/articles/glydet.html" class="external-link">Get
Started with glydet</a>.</p>
</div>
<div class="section level2">
<h2 id="whats-next">What’s Next?<a class="anchor" aria-label="anchor" href="#whats-next"></a>
</h2>
<p>This vignette has given you a taste of <code>glycoverse</code> in
action through a real-world glycomics workflow. But we’ve barely
scratched the surface! Now that you’ve got the basics down, you’re ready
to unlock the full potential of each package.</p>
<p>Here’s your roadmap to mastering each component:</p>
<ul>
<li>
<strong><a href="https://glycoverse.github.io/glyexp/articles/glyexp.html" class="external-link">glyexp</a></strong>
— Master experiment objects and data manipulation</li>
<li>
<strong><a href="https://glycoverse.github.io/glyread/articles/glyread.html" class="external-link">glyread</a></strong>
— Import and organize glycomics data</li>
<li>
<strong><a href="https://glycoverse.github.io/glyclean/articles/glyclean.html" class="external-link">glyclean</a></strong>
— Build custom preprocessing pipelines<br>
</li>
<li>
<strong><a href="https://glycoverse.github.io/glystats/articles/glystats.html" class="external-link">glystats</a></strong>
— Explore the full statistical toolkit</li>
<li>
<strong><a href="https://glycoverse.github.io/glyvis/articles/glyvis.html" class="external-link">glyvis</a></strong>
— Create stunning visualizations</li>
<li>
<strong><a href="https://glycoverse.github.io/glymotif/articles/glymotif.html" class="external-link">glymotif</a></strong>
— Define and analyze custom motifs</li>
<li>
<strong><a href="https://glycoverse.github.io/glydet/articles/glydet.html" class="external-link">glydet</a></strong>
— Create powerful derived traits</li>
<li>
<strong><a href="https://glycoverse.github.io/glyenzy/articles/glyenzy.html" class="external-link">glyenzy</a></strong>
— Explore enzyme-substrate relationships (we didn’t cover this one, but
it’s fascinating!)</li>
<li>
<strong><a href="https://glycoverse.github.io/glyrepr/articles/glyrepr.html" class="external-link">glyrepr</a></strong>
— Master glycan structure representation</li>
<li>
<strong><a href="https://glycoverse.github.io/glyparse/articles/glyparse.html" class="external-link">glyparse</a></strong>
— Parse and convert structural formats</li>
</ul>
<p>Happy glycan hunting! 🧬</p>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Bin Fu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
